apps/web/src/components/Example/Example.simple.stories.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import type { Meta, StoryObj } from "@storybook/react";
+
+import Example from "./Example";
+
+// The default export metadata controls how Storybook lists your stories and provides information used by addons.
+const meta: Meta<typeof Example> = {
+  component: Example,
+  // You can also define args at the component level; they will apply to all the component's stories unless you overwrite them. To do so, use the args key on the default CSF export:
+  argTypes: {
+    showBorder: { control: "boolean" },
+  },
+  args: {
+    showBorder: true,
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof Example>;
+
+// Use the named exports of a CSF file to define your component’s stories. We recommend you use UpperCamelCase for your story exports.
+export const WithDefaultArgs: Story = {
+  args: {
+    subtitle: "WithDefaultArgs",
+    color: "primary",
+  },
+};
+
+// A story can use a custom render function
+export const WithRenderFunction: Story = {
+  name: "With render function",
+  render: () => (
+    <Example subtitle="WithRenderFunction" color="primary" showBorder />
+  ),
+};
+
+export const WithArgCompositionAndParameters: Story = {
+  args: {
+    // combine args for multiple stories of the same component.
+    ...WithDefaultArgs.args,
+  },
+  parameters: {
+    // Parameters are a set of static, named metadata about a story, typically used to control the behavior of Storybook features and addons.
+    backgrounds: {
+      default: "white",
+      values: [{ name: "white", value: "#fff" }],
+    },
+    themeKey: "iap",
+  },
+};
```

Comment:
Yeah, it's true.  For example, in the `CustomFooter` story the non-nullable `color` prop is omitted and TS doesn't flag it.  The types are all partial because Storybook composes the final arg set from both the meta object and the individual story.

I found [this blog post](https://storybook.js.org/blog/improved-type-safety-in-storybook-7/) from a few months ago.  It describes how to use the `satisfies` operator instead of object typing to close this gap.  Confusingly, it seems to work in the IDE but refuses to compile for me.  I'll keep trying to get it working - it would be pretty useful.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/45580da0-772a-42cd-a0c0-98bac2478dc2)

Reply:
I found [this discussion](https://github.com/storybookjs/storybook/discussions/21587) that seemed promising.  It suggests using the component props type instead of typeof the component.  Alas, no luck.  :cry:

Reply:
From the same [blog post](https://storybook.js.org/blog/improved-type-safety-in-storybook-7/) @petertgiles referenced from February 8, 2023.

> Note, this codemod does not add the `satisfies` operator automatically. We plan to release a second codemod soon, which will enable users of TypeScript 4.9+ to migrate to the new types and add in the satisfies operator. Look out for that update in the coming weeks.

I found an issue that references the codemod but it is still not ready it seems: https://github.com/storybookjs/storybook/issues/20836

Reply:
Whoops, I see it too 😢.

## Screenshot
![Screen Shot 2023-10-24 at 10 40 54](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/5548df89-a762-42fd-a0cf-bfee146e1870)

Reply:
Took some time to look at this again and it seems like the offending code is the story that only includes a render function. I'm not sure why, but if you comment that specific story out everything builds fine 🤔 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e8684e030563c793a3024b399a5d25408192f850/apps/web/src/components/Example/Example.simple.stories.tsx#L30C1-L35C3

Reply:
@petertgiles what do you think, are you ok with removing the story that only includes a render function? do we have existing stories that are only a render function and if so, should we?

Reply:
Looks like we have 4 files that use the render function 👁️ 

![Screenshot 2023-11-08 115822](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/87d16846-7865-4df8-a5e3-69dee8591b57)

Reply:
maybe related https://github.com/storybookjs/storybook/discussions/21587#discussioncomment-5532108?

Reply:
There have been multiple updates to storybook since we last merged main in... I just came back to this and unless I'm missing something, it seems to have resolved itself? 🤔 

![Screenshot 2024-02-28 141552](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/928d5ba5-50d3-40b0-8cf2-b43238b290a4)